File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> numMap;

        for(int i= 0; i < nums.size(); i++){
            int index = target - nums[i];

            if(numMap.find(index) != numMap.end()){
                return{numMap[index], i };
            }
            numMap[nums[i]] = i;

        }
        return{};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        r , l = len(numbers) - 1, 0
        while l < r:
            sum_ = numbers[l] + numbers[r]
            if sum_ == target:
                return [l + 1,r + 1]
            elif sum_ > target:
                r -= 1
            else:
                l += 1
        return []   
